File: 1-easy/problems/0541-reverse-string-ii/solution.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        chars = list(s)
        for start in range(0, len(chars), 2 * k):
            chars[start:start + k] = reversed(chars[start:start + k])
        return "".join(chars)

File: 1-easy/problems/0541-reverse-string-ii/solution.ts
function reverseStr(s: string, k: number): string {
    const chars = s.split("");
    for (let start = 0; start < chars.length; start += 2 * k) {
        let left = start;
        let right = Math.min(start + k - 1, chars.length - 1);
        while (left < right) {
            [chars[left], chars[right]] = [chars[right], chars[left]];
            left++;
            right--;
        }
    }
    return chars.join("");
}
